### ISSUE TITLE
`onClick` warning when using a component that wraps `Button.Root` "as" another component that wraps `Button.Root`

### ISSUE THREAD

**Issue by fabien-ml** [fabien-ml/kobalte#62]
**Describe the bug**
This cause the `onClick` warning to appears on the button press :

```jsx
function IconButton(props: any) {
  return <Button.Root {...props} />
}

function App() {
  return <Dialog.CloseButton as={IconButton} />
}
```

**Expected behavior**
It should not display the warning.

**Comment by fabien-ml** [comment]
@riccardoperra

**Comment by fabien-ml** [comment]
fixed in `v0.4.0`.

**Comment by riccardoperra** [comment]
Thanks!
